### PR TITLE
Split up test_bad_function_pointer_cast using `@parameterized`. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
This only moved two the dimensions into parameters but its enough to make a big difference to the test precision.

File: tests/test_other.py [hunk 1]
```diff
@@ -4010,33 +4016,31 @@ def test_bad_function_pointer_cast(self):
...
-                # success!
-                self.assertContained('Hello, world.', output)
-              else:
-                # otherwise, the error depends on the mode we are in
-                # wasm trap raised by the vm
-                self.assertContained('function signature mismatch', output)
+      for emulate_casts in [0, 1]:
+        for relocatable in [0, 1]:
+          # wasm2js is not compatible with relocatable mode
+          if wasm == 0 and relocatable:
```
Comment (tlively) [inline tests/test_other.py:4022]:
```suggestion
          if not wasm and relocatable:
```

**Review by tlively — APPROVED** [review]
LGTM (disabled automerge due to comment)

File: tests/test_other.py (hunk 1 above)
Comment (sbc100) [reply tests/test_other.py:4022]: Oops, good call.  Done.

**Pull request merged**